src/osd/PG.cc

```diff
@@ -3847,6 +3850,30 @@ void PG::replica_scrub(
     return;
   }
 
+  scrubber.active_rep_scrub = op;
+  scrubber.map_state = PG::Scrubber::REG_SCRUBBING;
+  osd->queue_for_build_scrub_map(this, g_conf->osd_scrub_map_start_cost);
+}
+
+void PG::replica_scrub(epoch_t queued, ThreadPool::TPHandle &handle)
+{
+  dout(20) << __func__ << " state " << Scrubber::map_state_string(scrubber.map_state)
+           << " objects " << scrubber.ls.size() << dendl;
+  assert(_lock.is_locked());
+  assert(scrubber.active_rep_scrub);
```

Comment:
As I mentioned earlier, if you want this to be set at this point, you'll probably need to adjust the existing users.